### ISSUE TITLE
Long MTConnect Device UUID and Name - stack smashing detected

### ISSUE THREAD

**Issue by fastxl** [mtconnect/dataminer#3]
Trying to set this up with one of our machine tools that has MTConnect installed. When I run

`dataminter /tmp http://MachineIP:5000`

I get 
```
output: [ MyMachineTool-20201117-091004.log - records: 60 ]
*** stack smashing detected ***: terminated
Aborted (core dumped)
```
when I use it with a sample agent and adapter I do NOT get this issue. Seems to work. Any suggestions?

**Comment by simonyg** [comment]
What's OS? Do you have the tool to show the stack trace from the core dump file?

Check the content of /tmp/MyMachineTool-20201117-091004.log. It should have 60 records. Does it look right?

**Comment by fastxl** [comment]
OS is Ubuntu Server 20.04.1

Tool to show Stack Trace??? Not sure about this.

The data in /tmp/MyMachineTool-20201117-091004.log has 60 records in it and seems to be correct.

No dataminer.settings file was created.

**Comment by simonyg** [comment]
It's good. that the data log file is correct. That means it is able to process the initial data returned from your MTConnect agent. 

A core dump file (**core***) should be generated in the directory you run dataminer. You can show the stack trace using gdb with "bt" command:

https://stackoverflow.com/questions/7848771/how-can-one-see-content-of-stack-with-gdb

What's your output when running dataminer? Please mask out any specific content. It should look like this:

Restoring - https://smstestbed.nist.gov/vds/GFAgie01 = 95898739|2020/09/20-00:04:18
----------------
Output Location: /tmp
Agent Uri:       http://mtconnect.mazakcorp.com:5610
Poll Interval:   60
Next Sequence #: 
output: [ MTConnect-MAZAK-M77KP290337-MFMS10-MC2-20201117-074917.log - records: 94 ]
========== { http://mtconnect.mazakcorp.com:5610 - 1, next sequence = 512637 } ==========

What is your Ubuntu arch? ARM, AMD or Intel. You can find out by running "uname -a".

**Comment by fastxl** [comment]
I am running dataminer from my home directory and I do not get the core dump file.

When I run  `dataminer /tmp https://smstestbed.nist.gov/vds` which is the sample nist data

I get this output and it seems to continue writing log files
```
Output Location: /tmp
Agent Uri:       https://smstestbed.nist.gov/vds
Poll Interval:   60
Next Sequence #:
output: [ MTConnect-mtc_adapter001-GFAgie01-20201117-111143.log - records: 25 ]
output: [ MTConnect-mtc_adapter005-Hurco01-20201117-111143.log - records: 17 ]
output: [ MTConnect-mtc_adapter006-Hurco02-20201117-111143.log - records: 17 ]
output: [ MTConnect-mtc_adapter007-Hurco03-20201117-111143.log - records: 17 ]
output: [ MTConnect-mtc_adapter008-Hurco04-20201117-111143.log - records: 17 ]
output: [ MTConnect-mtc_adapter010-Hurco06-20201117-111143.log - records: 17 ]
output: [ MTConnect-mtc_adapter002-Mazak01-20201117-111143.log - records: 67 ]
output: [ MTConnect-mtc_adapter004-Mazak03-20201117-111143.log - records: 65 ]
========== { https://smstestbed.nist.gov/vds - 1, next sequence = 206113490 } ==========
Updating - https://smstestbed.nist.gov/vds = 206113490
```

If I run `dataminer /tmp http://MyMachineIP:5000` I get my original core dump error

`uname -a` results are this
`Linux MyHostName 5.4.0-42-generic #46-Ubuntu SMP Fri Jul 10 00:24:02 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by simonyg** [comment]
*** stack smashing detected ***: terminated

it is a stack overflow problem. Likely the data returned from your MTConnect agent is larger than expected. To pinpoint where the problem is, core dump file will help a lot. Set core dump file size to be unlimited:

```
ulimit -c unlimited 
ulimit -a 
```

And run it again. The core file may be in your /tmp.

If this fails, I can send you a new executable with more trace output.

**Comment by fastxl** [comment]
Ok, got the core file and ran this `gdb dataminer core`

This is what I got after running `bt`

```
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007fdad357d859 in __GI_abort () at abort.c:79
#2  0x00007fdad35e83ee in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7fdad371207c "*** %s ***: terminated\n") at ../sysdeps/posix/libc_fatal.c:155
#3  0x00007fdad368ab4a in __GI___fortify_fail (msg=msg@entry=0x7fdad3712064 "stack smashing detected") at fortify_fail.c:26
#4  0x00007fdad368ab16 in __stack_chk_fail () at stack_chk_fail.c:24
#5  0x0000564deb21e4d9 in agentHandler::outputJSON(Settings*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) ()
#6  0x0000564deb2baaf8 in Worker::poll() ()
#7  0x0000564deb2ba52d in Worker::run() ()
#8  0x0000564deb21b6b3 in boost::_mfi::mf0<void, Worker>::operator()(Worker*) const ()
#9  0x0000564deb21b5ab in void boost::_bi::list1<boost::_bi::value<Worker*> >::operator()<boost::_mfi::mf0<void, Worker>, boost::_bi::list0>(boost::_bi::type<void>, boost::_mfi::mf0<void, Worker>&, boost::_bi::list0&, int) ()
#10 0x0000564deb21b12e in boost::_bi::bind_t<void, boost::_mfi::mf0<void, Worker>, boost::_bi::list1<boost::_bi::value<Worker*> > >::operator()() ()
#11 0x0000564deb21aa8e in boost::detail::thread_data<boost::_bi::bind_t<void, boost::_mfi::mf0<void, Worker>, boost::_bi::list1<boost::_bi::value<Worker*> > > >::run() ()
#12 0x00007fdad3d1143b in ?? () from /lib/x86_64-linux-gnu/libboost_thread.so.1.71.0
#13 0x00007fdad3c35609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#14 0x00007fdad367a293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Comment by simonyg** [comment]
By any chance, does your MTConnect agent return a long string of UUID or Device name?

```
curl https://smstestbed.nist.gov/vds/GFAgie01/probe
...
    <Device id="GFAgie01" name="GFAgie01" uuid="mtc_adapter001">
...
```

**Comment by fastxl** [comment]
Yes, I would say it's rather long

` <Device id="OKUMA.MachiningCenterMA650-EAST.190056" name="OKUMA.MachiningCenterMA650-EAST" uuid="OKUMA.MachiningCenterMA650-EAST.190056">`

**Comment by simonyg** [comment]
[dataminer.zip](https://github.com/mtconnect/dataminer/files/5555711/dataminer.zip)

That may be the issue. I have removed the fixed buffer size. Try this.

**Comment by fastxl** [comment]
Seems to be a winner thank you for the help. Now to work on the ELK part

**Comment by simonyg** [comment]
This is assigned to be officially fixed in v1.1.2.

**Comment by Icydeath** [comment]
I'm running into the same issue (long UUID and Name). I know you said you were working on putting this in the official release of 1.1.2 but it's been some time now. Any chance you can share a compiled windows version that resolves this problem?

Your effort and time is much appreciated.

**Comment by simonyg** [comment]
Released v1.1.2. Please check.

**Comment by Icydeath** [comment]
much appreciated. I'll give it a go and report back.
